src/encoded/types/functional_characterization_experiment.py

```diff
@@ -234,6 +297,36 @@ def datapoint(self, request, related_series=None):
                 return True 
         return False
 
+    @calculated_property(schema={
+        "title": "Biosamples",
+        "type": "array",
+        "items": {
+            "type": 'string',
+            "linkTo": "Biosample",
+        },
+        "notSubmittable": True,
+    })
+    def biosamples(self, request, replicates=None):
+        biosamples = set()
+        if replicates:
+            properties = {'replicates': replicates}
+            path = Path(
+                'replicates.library.biosample', 
+                include=[
+                    '@id',
+                    'library',
+                    'biosample',
+                ]
+            )
+            path.expand(request, properties)
+            
+            if properties['replicates']:
```

Comment:
This seems redundant as you have just assigned properties 2 actions above.  Maybe I am missing a logic flow.

Reply:
just trying to avoid path.expand resulting in 'replicates' pointing to None, maybe it is not going to happen, but better safe than throwing exception trying to iterate over None.